sql-cli/noxfile.py

```diff
@@ -21,15 +21,48 @@ def dev(session: nox.Session) -> None:
 
 
 @nox.session(python=["3.7", "3.8", "3.9"])
-def test(session: nox.Session) -> None:
+@nox.parametrize("airflow", ["2.2.5", "2.3.4", "2.4.2"])
+def test(session: nox.Session, airflow: str) -> None:
     """Run both unit and integration tests."""
+
     session.install("poetry")
-    session.run("poetry", "install", "--with", "dev")
-    # Log all the installed dependencies
+    if airflow.startswith("2.1") or airflow.startswith("2.2"):
+        # To install some versions of Airflow, we need constraints, due to issues like:
+        # https://github.com/apache/airflow/issues/19804
+        constraints_url = (
+            "https://raw.githubusercontent.com/apache/airflow/"
+            f"constraints-{airflow}/constraints-{session.python}.txt"
+        )
+        session.run("rm", "-rf", "dist", external=True)
+
+        # Poetry does not support constraints:
+        # https://github.com/python-poetry/poetry/issues/3225
+        # We attempted a few approaches to get this to work, but they were unsuccessful.
+        session.run("poetry", "build", "-f", "wheel")
+        session.install("dist/astro_sql_cli-0.1.1-py3-none-any.whl", "-c", constraints_url)
+
+        # We are duplicating the tests dependencies until we find a better solution.
+        # The solution might be to move out of poetry.
+        session.install("pytest")
+        session.install("pytest-cov")
+        session.install("mypy")
+        session.install("types-pyyaml")
+    else:
+        session.run("poetry", "install", "--with", "dev")
+        session.run("poetry", "run", "pip", "install", f"apache-airflow=={airflow}")
+
     session.log("Installed Dependencies:")
     session.run("pip3", "freeze")
-    airflow_home = f"~/airflow-latest-{session.python}"
-    session.run("airflow", "db", "init", env={"AIRFLOW_HOME": airflow_home})
+
+    # At the moment flow run depends on the Airflow global Airflow Home - we need to fix this
```

Comment:
Would there be a way to fix this now or is this a bigger fix? At minimum I'd like to see a ticket tied to this.

Reply:
@dimberman it's a bigger fix and I've logged a dedicated ticket for this last week:
https://github.com/astronomer/astro-sdk/issues/1219